src/cryptography.rs

```diff
@@ -72,12 +73,13 @@ pub fn create_key_pair() -> KeyPair {
     }
 }
 
-pub fn create_default_hash(value: &Value) -> Result<Vec<u8>> {
+pub fn create_default_hash(value: &Value) -> Result<Vec<u8>, Error> {
     create_hash(value, &HashAlgorithm::default())
 }
 
-pub fn create_hash(value: &Value, hash_algorithm: &HashAlgorithm) -> Result<Vec<u8>> {
-    let canonical_json = serde_jcs::to_vec(value)?;
+pub fn create_hash(value: &Value, hash_algorithm: &HashAlgorithm) -> Result<Vec<u8>, Error> {
+    let canonical_json =
+        serde_jcs::to_vec(value).map_err(|e| Error::JsonDoesNotConformToSchema(e.to_string()))?;
```

Comment:
This error here is not a `JsonDoesNotConformToSchema` error, it is an error during serialization (which probably will never happen).
I think it would be better to have a separate error for this: `JcsSerializationError` or something similar.